cuML/src/knn/knn.cu

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2018, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "knn_c.h"
+#include <faiss/IndexFlat.h>
+#include <faiss/gpu/GpuIndexFlat.h>
+#include <faiss/gpu/GpuIndexIVFFlat.h>
+#include <faiss/gpu/StandardGpuResources.h>
+
+namespace ML {
+
+using namespace MLCommon;
+
+class kNN {
+
+    faiss::gpu::GpuIndexFlatL2 index_flat;
+    
+    public:
+
+	kNN::kNN() {}
+	kNN::~kNN() {}
+
+        void kNN::fit(float *input, int N, int D) {
+            
+	    faiss::gpu::StandardGpuResources res;
```

Comment:
It's good to have this confirmed because I was noticing this behavior and made the changes in a more recent PR: https://github.com/rapidsai/cuml/pull/64

I've got the D2D transfer from cuDF working.